packages/cli/src/commands/integration-resource/create-threshold.ts

```diff
@@ -0,0 +1,366 @@
+import { parseArguments } from '../../util/get-args';
+import { getFlagsSpecification } from '../../util/get-flags-specification';
+import { getResources } from '../../util/integration-resource/get-resources';
+import { IntegrationResourceCreateThresholdTelemetryClient } from '../../util/telemetry/commands/integration-resource/create-threshold';
+import { createThresholdSubcommand } from './command';
+import getScope from '../../util/get-scope';
+import { getBalanceInformation } from '../../util/integration/fetch-installation-prepayment-info';
+import { updateThreshold } from '../../util/integration-resource/update-threshold';
+import { updateInstallationThreshold } from '../../util/integration/update-installation-threshold';
+import type {
+  CreditWithAmount,
+  InstallationBalancesAndThresholds,
+  PrepaymentCreditThreshold,
+} from '../../util/integration/types';
+import type { Resource } from '../../util/integration-resource/types';
+
+export async function createThreshold(client: Client) {
+  const telemetry = new IntegrationResourceCreateThresholdTelemetryClient({
+    opts: {
+      store: client.telemetryEventStore,
+    },
+  });
+
+  let parsedArguments = null;
+  const flagsSpecification = getFlagsSpecification(
+    createThresholdSubcommand.options
+  );
+
+  try {
+    parsedArguments = parseArguments(client.argv.slice(3), flagsSpecification);
+  } catch (error) {
+    printError(error);
+    return 1;
+  }
+
+  // parse arguments
+  let args: ReturnType<typeof parseCreateThresholdArguments>;
+
+  try {
+    args = parseCreateThresholdArguments(parsedArguments.args, telemetry);
+  } catch (error: unknown) {
+    output.error((error as Error).message);
+    return 1;
+  }
+
+  const skipConfirmWithYes = parsedArguments.flags['--yes'];
+  telemetry.trackCliFlagYes(skipConfirmWithYes);
+
+  const { resourceName, minimum, spend, limit } = args;
+
+  // Fetch Team
+  const { team } = await getScope(client);
+  if (!team) {
+    output.error('Team not found.');
+    return 1;
+  }
+
+  // Fetch Resource
+  output.spinner('Retrieving resource…', 500);
+  const resources = await getResources(client, team.id);
+  const targetedResource = resources.find(
+    resource => resource.name === resourceName
+  );
+  output.stopSpinner();
+
+  // Assert resource is valid
+  if (!targetedResource) {
+    output.log(`The resource ${chalk.bold(resourceName)} was not found.`);
+    return 0;
+  }
+
+  if (!targetedResource.product?.integrationConfigurationId) {
+    output.error(
+      `The resource ${chalk.bold(resourceName)} does not have an integration configuration.`
+    );
+    return 1;
+  }
+
+  if (targetedResource.billingPlan?.type !== 'prepayment') {
+    output.error(
+      `The resource ${chalk.bold(resourceName)} is not a prepayment-based resource.`
+    );
+    return 1;
+  }
+
+  // Assert spend is within billing plan limits
+  const minimumSpend =
+    parseFloat(targetedResource.billingPlan.minimumAmount ?? '0.50') * 100;
```

Comment:
I assume these match some values also validated in the API? (not blocking)

Reply:
Yeah, this is a little minimum-amount limit we generally apply to avoid 1 cent purchases and all that. It's validated on the API layer, but I'm doing some fast-failure here to provide a useful error message and save some wait time in execution.

Reply:
Just wanted to be sure we had some defense in depth on this. I wish we had a good pattern for sharing these values. I've seen them drift over time, but that's not a problem to solve today!